### ISSUE TITLE
E2541 is not working properly

### ISSUE THREAD

**Issue by vbarba** [awslabs/cfn-python-lint#405]
*cfn-lint version: 0.8.1*

*Description of issue.*

E2541 For all currently supported action types, the only valid version string is "1".

But

> Version: "1"

**Comment by kddejong** [comment]
I looked into this quick and wasn't able to reproduce.  I was hoping to not see more of your template but I may need to do this justice.

In our testing (and testing manually) this template https://github.com/awslabs/cfn-python-lint/blob/master/test/fixtures/templates/good/resources_codepipeline.yaml is coming back clean which does have the `Version: "1"` set.  I flipped it to "2" just to check and it did trigger.  The only other thing I see here is that we are making sure its a string and not an integer.  Your example shouldn't parse as an integer though.

**Comment by vbarba** [comment]
I am sorry, can you please double-check?

Here the Codepipeline resource code:


    Pipeline:
      Type: AWS::CodePipeline::Pipeline
      Properties:
        Name: !Sub Payments-${Environment}
        RoleArn: !GetAtt CodePipelineServiceRole.Arn
        ArtifactStore:
          Type: S3
          Location: !Ref ArtifactBucket
        Stages:
          - Name: Source
            Actions:
              - Name: Payments
                ActionTypeId:
                  Category: Source
                  Provider: S3
                  Owner: AWS
                  Version: "1"
                Configuration:
                  S3Bucket: wallapop-trigger-link-cicd
                  S3ObjectKey: !Sub payment/payment-${Tag}.zip
                OutputArtifacts:
                  - Name: App
          - Name: Infrastructure
            Actions:
              - Name: RDS
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: CloudFormation
                Configuration:
                  ActionMode: CREATE_UPDATE
                  Capabilities: CAPABILITY_IAM
                  StackName: !Sub PaymentsRDS-${Environment}
                  TemplateConfiguration: !Sub App::infrastructure/parameters/payments/rds/${Environment}.json
                  TemplatePath: App::infrastructure/templates/payments/payments-rds.yml
                  RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                InputArtifacts:
                  - Name: App
              - Name: Resources
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: CloudFormation
                Configuration:
                  ActionMode: CREATE_UPDATE
                  Capabilities: CAPABILITY_IAM
                  StackName: !Sub Payments-${Environment}
                  TemplateConfiguration: !Sub App::infrastructure/parameters/payments/resources/${Environment}.json
                  TemplatePath: App::infrastructure/templates/payments/payments-resources.yml
                  RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                InputArtifacts:
                  - Name: App
              - Name: Service
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: CloudFormation
                Configuration:
                  ActionMode: CREATE_UPDATE
                  Capabilities: CAPABILITY_IAM
                  StackName: !Sub payments-service-${Environment}
                  TemplateConfiguration: !Sub App::infrastructure/parameters/payments/service/${Environment}.json
                  TemplatePath: App::infrastructure/templates/payments/payments-service.yml
                  RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                InputArtifacts:
                  - Name: App
              - Name: Pipeline
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: CloudFormation
                Configuration:
                  ActionMode: CREATE_UPDATE
                  Capabilities: CAPABILITY_IAM
                  StackName: !Sub payments-pipeline-${Environment}
                  TemplateConfiguration: !Sub App::infrastructure/parameters/payments/pipeline/${Environment}.json
                  TemplatePath: App::infrastructure/templates/payments/payments-pipeline.yml
                  RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                InputArtifacts:
                  - Name: App
              - Name: Alarms
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: CloudFormation
                Configuration:
                  ActionMode: CREATE_UPDATE
                  Capabilities: CAPABILITY_IAM
                  StackName: !Sub payments-alarms-${Environment}
                  TemplateConfiguration: !Sub App::infrastructure/parameters/payments/alarms/${Environment}.json
                  TemplatePath: App::infrastructure/templates/payments/payments-alarms.yml
                  RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                InputArtifacts:
                  - Name: App
              - !If
                - CreateProdResources
                - Name: Dashboard
                  ActionTypeId:
                    Category: Deploy
                    Owner: AWS
                    Version: "1"
                    Provider: CloudFormation
                  Configuration:
                    ActionMode: CREATE_UPDATE
                    Capabilities: CAPABILITY_IAM
                    StackName: !Sub payments-dashboard-${Environment}
                    TemplateConfiguration: !Sub App::infrastructure/parameters/payments/dashboard/${Environment}.json
                    TemplatePath: App::infrastructure/templates/payments/payments-dashboard.yml
                    RoleArn: !GetAtt CloudFormationExecutionRole.Arn
                  InputArtifacts:
                    - Name: App
                - !Ref AWS::NoValue
          - Name: Build
            Actions:
            - Name: Flyway
              ActionTypeId:
                Category: Build
                Owner: AWS
                Version: "1"
                Provider: CodeBuild
              Configuration:
                ProjectName: !Ref CodeBuildDBScripts
              InputArtifacts:
              - Name: App
          - Name: Deploy
            Actions:
              - Name: Deploy
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Version: "1"
                  Provider: ECS
                Configuration:
                  ClusterName: !Ref Environment
                  ServiceName:
                      Fn::ImportValue:
                        !Sub "payments-${Environment}-name"
                  FileName: images.json
                InputArtifacts:
                  - Name: App
                RunOrder: 1`

If I run the last version linter:
```
$ sudo pip install -U cfn-lint
$ cfn-lint -t infrastructure/templates/delivery/delivery-pipeline.yml
E2541 For all currently supported action types, the only valid version string is "1".
infrastructure/templates/delivery/delivery-pipeline.yml:391:13
```

If I run an old version:

```
$ sudo pip install -U cfn-lint==0.6.1
$ cfn-lint -t infrastructure/templates/delivery/delivery-pipeline.yml
$
```

**Comment by kddejong** [comment]
ok I was able to replicate this now.  I'm looking into it.
